include/drivers/gpio.h

```diff
@@ -1217,19 +1230,15 @@ static inline int gpio_pin_toggle(struct device *port, unsigned int pin)
  * @param value Value set on the pin.
  * @return 0 if successful, negative errno code on failure.
  *
- * @deprecated Replace with gpio_pin_set_raw() or gpio_pin_set().
+ * @deprecated Replace with gpio_pin_set_raw(), or gpio_pin_set() if
+ * active level is handled correctly.
  */
-static inline int gpio_pin_write(struct device *port, u32_t pin,
-				 u32_t value)
+/* Deprecated in 2.2 release */
+__deprecated static inline int gpio_pin_write(struct device *port,
+					      gpio_pin_t pin,
+					      u32_t value)
 {
-	const struct gpio_driver_config *const cfg =
-		(const struct gpio_driver_config *)port->config->config_info;
-
-	(void)cfg;
-	__ASSERT((cfg->port_pin_mask & (gpio_port_pins_t)BIT(pin)) != 0U,
-		 "Unsupported pin");
-
-	return gpio_write(port, GPIO_ACCESS_BY_PIN, pin, value);
+	return gpio_pin_set_raw(port, pin, value != 0);
```

Comment:
Several of the existing GPIO drivers support GPIO_POL_INV flag. An examples are: gpio_cc13xx_cc26xx.c, gpio_esp32.c, gpio_mmio32.c, gpio_pcal9535a.c, gpio_sifive.c, gpio_sx1509b.c, gpio_nrfx.c. The effect of passing the GPIO_POL_INV flag is inversion of the pin level. If the GPIO_POL_INV flag was passed to the `gpio_pin_configure` function subsequent calling of `gpio_pin_write(,,1)` will result in pin being driven low. That behavior can only be correctly implemented with help of `gpio_pin_set`. Any out of tree application that is currently using GPIO_POL_INV flag and one of the affected drivers will break if `gpio_pin_set_raw` is used. We need to preserve backwards compatibility.